raftLogCheck.js

```diff
@@ -0,0 +1,235 @@
+const http = require('http');
+const async = require('async');
+const md5 = require('md5');
+const werelogs = require('werelogs');
+
+const logLevel = Number.parseInt(process.env.DEBUG, 10) === 1
+    ? 'debug' : 'info';
+const loggerConfig = {
+    level: logLevel,
+    dump: 'error',
+};
+werelogs.configure(loggerConfig);
+const log = new werelogs.Logger('s3utils::raftLogCheck');
+
+let pageSize = process.env.PAGE_SIZE;
+if (pageSize === undefined) {
+    pageSize = 1000;
+}
+
+const myArgs = process.argv.slice(2);
+
+if (myArgs.length !== 3) {
+    log.error(
+        'usage: node raftLogCheck.js bucketdHost bucketdPort raftSession');
+    process.exit(1);
+}
+
+const bucketdHost = myArgs[0];
+const bucketdPort = myArgs[1];
+const raftSession = myArgs[2];
+
+function getUrl(url, cb) {
+    log.debug('url', { url });
+    http.get(url, res => {
+        let body = '';
+
+        res.on('data', chunk => {
+            body += chunk;
+        });
+
+        res.on('end', () => {
+            try {
+                return cb(null, JSON.parse(body));
+            } catch (err) {
+                return cb(err);
+            }
+        });
+    }).on('error', err => (cb(err))
+    );
+}
+
+function compareWithLeaderLog(leaderLog, connected, seq, cb) {
+    log.debug('leaderLog:', { leaderLog });
+    async.each(connected, (member, cb) => {
+        getUrl(
+            `http://${member.host}:${member.adminPort}` +
+            `/_/raft/log?begin=${seq}&limit=${pageSize}`,
+            (err, memberLog) => {
+                if (err) {
+                    return cb(err);
+                }
+                log.info('getting member page', { member, seq });
```

Comment:
Create and use a request logger here and end it at the end of the iteratee function, if a log dump happens because of an error it will dump all logs and may crash the process.